cli/src/semgrep/engine.py

```diff
@@ -63,7 +63,8 @@ def get_cpu_count() -> int:
     def default_jobs(self) -> int:
         if self == EngineType.PRO_INTERFILE:
             return 1
-        return self.get_cpu_count()
+
```

Comment:
Please add a comment in the code why we use a max of 16 (mostly copy what you put in your PR explanation in the code instead).